### PR TITLE
Add new config type CFG_BYTES, and use it for trace_id and span_id

### PR DESCRIPTION
**Pull request by cydrain**
Issue: https://github.com/milvus-io/milvus/issues/30961

### CODE REVIEW

**Comment by alexanderguzhva** [comment]
@cydrain Why CFG_STRING is insufficient for this purpose?

**Comment by cydrain** [comment]
> @cydrain Why CFG_STRING is insufficient for this purpose?

you can see https://github.com/milvus-io/milvus/issues/30961, convert an arbitrary binary into a string and save it into search config, there may be problem when call config.dump().
Saving a binary as a uint8_t vector, can avoid such kind of issue.

File: include/knowhere/config.h [hunk 1]
```diff
@@ -57,6 +58,10 @@ typedef nlohmann::json Json;
 #define CFG_MATERIALIZED_VIEW_SEARCH_INFO_TYPE std::optional<MaterializedViewSearchInfo>
 #endif
 
+#ifndef CFG_VECTOR
+#define CFG_VECTOR std::optional<std::vector<uint8_t>>
```
Comment (zhengbuqian) [inline include/knowhere/config.h:62]: use `CFG_BINARY` or `CFG_BYTES`

File: src/common/comp/brute_force.cc [hunk 2]
```diff
@@ -63,10 +63,9 @@ BruteForce::Search(const DataSetPtr base_dataset, const DataSetPtr query_dataset
 #ifdef NOT_COMPILE_FOR_SWIG
     std::shared_ptr<tracer::trace::Span> span = nullptr;
     if (cfg.trace_id.has_value()) {
-        auto trace_id_str = tracer::HexToString(cfg.trace_id.value());
-        auto span_id_str = tracer::HexToString(cfg.span_id.value());
-        auto ctx = tracer::TraceContext{(uint8_t*)trace_id_str.c_str(), (uint8_t*)span_id_str.c_str(),
-                                        (uint8_t)cfg.trace_flags.value()};
+        auto trace_id = cfg.trace_id.value();
+        auto span_id = cfg.span_id.value();
+        auto ctx = tracer::TraceContext{trace_id.data(), span_id.data(), (uint8_t)cfg.trace_flags.value()};
```
Comment (zhengbuqian) [inline src/common/comp/brute_force.cc:68]: can we avoid repeating these code?

Comment (cydrain) [reply src/common/comp/brute_force.cc]: done

File: include/knowhere/config.h (hunk 1 above)
Comment (cydrain) [reply include/knowhere/config.h]: updated

**Pull request merged**